src/workflows/airqo_etl_utils/airqo_utils.py

```diff
@@ -370,15 +373,15 @@ def extract_devices_data(
                         )
                     )
 
-                data["device_number"] = device_number
-                data["device_id"] = device.get("device_id")
-                data["site_id"] = device.get("site_id")
+                meta_data = data.attrs.get("meta_data", {})
```

Comment:
Do you know why the data was being stripped of the `meta_data` in the earlier code version? I see that you are getting it now and not popping it.